### PR TITLE
chore: specialize build envs

### PR DESCRIPTION
**Pull request by jyeshe**
- Prepare for Elixir releases with explicit prod build env (runtime will be from `runtime.exs`)
- Reintroduce `AETERNITY_CONFIG` env var as it's needed by node on runtime

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 set -xe
 
-elixir --sname aeternity@localhost -S mix test
+MIX_ENV=test elixir --sname aeternity@localhost -S mix test
```
Comment (sborrazas) [inline scripts/test.sh:5]: This isn't really needed, since it's set by `mix test` automatically

Comment (jyeshe) [reply scripts/test.sh]: done, made implicit again

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ services:
     build:
       context: .
       dockerfile: ./Dockerfile
+      args:
+        MIX_ENV: dev
+    image: aeternity/ae_mdw_dev:latest
```
Comment (venimus) [inline docker-compose-dev.yml:9]: we don't have this repository `ae_mdw_dev` in dockerhub

Comment (venimus) [reply docker-compose-dev.yml:9]: I guess that needs a rebase

**Review by venimus — APPROVED** [review]

**Pull request merged**